cmd/nerdctl/compose.go

```diff
@@ -174,17 +178,38 @@ func getComposer(cmd *cobra.Command, client *containerd.Client) (*composer.Compo
 			ocispecPlatforms = []ocispec.Platform{parsed} // no append
 		}
 		var imgErr error
+		// IPFS reference
 		if scheme, ref, err := referenceutil.ParseIPFSRefWithScheme(imageName); err == nil {
 			ipfsClient, err := httpapi.NewLocalApi()
 			if err != nil {
 				return err
 			}
 			_, imgErr = ipfs.EnsureImage(ctx, client, ipfsClient, cmd.OutOrStdout(), cmd.ErrOrStderr(), snapshotter, scheme, ref,
 				pullMode, ocispecPlatforms, nil, quiet)
-		} else {
-			_, imgErr = imgutil.EnsureImage(ctx, client, cmd.OutOrStdout(), cmd.ErrOrStderr(), snapshotter, imageName,
-				pullMode, insecure, hostsDirs, ocispecPlatforms, nil, quiet)
+			return imgErr
+		}
+
+		ref := imageName
+		if verifier, ok := ps.Unparsed.Extensions[serviceparser.ComposeVerify]; ok {
+			switch verifier {
+			case "cosign":
+				keyRef, ok := ps.Unparsed.Extensions[serviceparser.ComposeCosignPublicKey]
+				if !ok {
+					return fmt.Errorf("no cosign public key, service: %s", ps.Unparsed.Name)
+				}
+
+				ref, err = cosignutil.VerifyCosign(ctx, ref, keyRef.(string), hostsDirs)
+				if err != nil {
+					return err
+				}
+			case "none":
+				logrus.Debugf("verification process skipped")
+			default:
+				return fmt.Errorf("no verifier found: %s", verifier)
+			}
 		}
+		_, imgErr = imgutil.EnsureImage(ctx, client, cmd.OutOrStdout(), cmd.ErrOrStderr(), snapshotter, ref,
+			pullMode, insecure, hostsDirs, ocispecPlatforms, nil, quiet)
```

Comment:
This can simply be `_, err := imgutil.EnsureImage(...` and `return err` maybe

Reply:
fixed.